src/shared/utilities/vsCodeUtils.ts

```diff
@@ -116,3 +116,8 @@ export function getChannelLogger(channel: vscode.OutputChannel): ChannelLogger {
             }),
     })
 }
+
+export function isFileIconThemeSeti(): boolean {
+    const iconTheme = vscode.workspace.getConfiguration('workbench').get('iconTheme')
```

Comment:
Double checked the behavior of `vscode.workspace.getConfiguration`. It will always return a configuration object even if the configuration does not exist. It acts like an empty configuration. So looks like we're always safe to chain these kinds of lookups.

Reply:
That's the implementation today, but it could change at a later time. Leaving it to your discretion though.

Reply:
Yeah, based on their doc, I would consider that a breaking change.

`@return The full configuration or a subset.`
And makes no mention of errors thrown.

Even if it weren't as clear in the doc, changing the behavior of any public API should be considered a breaking change. Users can and will code against the observed behavior at the time of implementation.

---

This reminds me of an example when we had a service that said its API could return any value in the given enum. But in practice it only ever returned X or Y. We made a behavior change where it started to return Z. At least 1 client was unable to handle Z and it turned into a much larger change and rollout process than we anticipated.

Moral was basically that you really need to advertise and prepare clients for behavior changes, even if the behavior is within the specification (or the behavior was previously undefined). Even if you explicitly state that something is possible, if it doesn't occur in practice and you don't explicitly state how to interpret the outcome, clients will not (and can't reasonably be expected to) code for that specific scenario.